libs/langchain/langchain/chat_models/mlflow.py

```diff
@@ -0,0 +1,206 @@
+import asyncio
+import logging
+from functools import partial
+from typing import Any, Dict, List, Mapping, Optional
+from urllib.parse import urlparse
+
+from langchain_core.callbacks import (
+    AsyncCallbackManagerForLLMRun,
+    CallbackManagerForLLMRun,
+)
+from langchain_core.language_models import BaseChatModel
+from langchain_core.messages import (
+    AIMessage,
+    BaseMessage,
+    ChatMessage,
+    FunctionMessage,
+    HumanMessage,
+    SystemMessage,
+)
+from langchain_core.outputs import ChatGeneration, ChatResult
+from langchain_core.pydantic_v1 import BaseModel, Extra, PrivateAttr
+
+logger = logging.getLogger(__name__)
+
+
+# Ignoring type because below is valid pydantic code
+# Unexpected keyword argument "extra" for "__init_subclass__" of "object"  [call-arg]
+class ChatParams(BaseModel, extra=Extra.allow):  # type: ignore[call-arg]
+    """Parameters for the `Mlflow` LLM."""
+
+    temperature: float = 0.0
+    n: int = 1
+    stop: Optional[List[str]] = None
+    max_tokens: Optional[int] = None
+
+
+class ChatMlflow(BaseChatModel):
+    """`MLflow` chat models API.
+
+    To use, you should have the `mlflow[genai]` python package installed.
+    For more information, see https://mlflow.org/docs/latest/llms/deployments/server.html.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.chat_models import ChatMlflow
+
+            chat = ChatMlflow(
+                target_uri="http://localhost:5000",
+                endpoint="chat",
+                params={"temperature": 0.1},
+            )
+    """
+
+    endpoint: str
+    """The endpoint to use."""
+    target_uri: str
+    """The target URI to use."""
+    params: Optional[ChatParams] = None
+    """Extra parameters such as `temperature`."""
+    _client: Any = PrivateAttr()
+
+    def __init__(self, **kwargs: Any):
+        super().__init__(**kwargs)
+        self._validate_uri()
+        try:
+            from mlflow.deployments import get_deploy_client
+
+            self._client = get_deploy_client(self.target_uri)
+        except ImportError as e:
+            raise ImportError(
+                "Failed to create the client. "
+                f"Please run `pip install mlflow{self._mlflow_extras}` to install "
+                "required dependencies."
+            ) from e
+
+    @property
+    def _mlflow_extras(self) -> str:
+        return "[genai]"
+
+    def _validate_uri(self) -> None:
+        if urlparse(self.target_uri).scheme not in ("http", "https"):
+            raise ValueError(
+                "Invalid target URI. The target URI must be a valid HTTP/HTTPS URI."
+            )
```

Comment:
We should allow `target_uri="databricks"`. `MlflowChat`, `Mlflow`, `MlflowEmbeddings` should still work with Databricks. Can we update this and test?

Reply:
I see, then I'll remove the validation.